src/Build.UnitTests/Graph/ProjectGraph_Tests.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Evaluation;
+using Xunit;
+
+namespace Microsoft.Build.Graph.UnitTests
+{
+    public class ProjectGraphTests
+    {
+        [Fact]
+        public void TestGraphWithSingleNode()
+        {
+            string projectContents = @"
+                <Project xmlns='http://schemas.microsoft.com/developer/msbuild/2003'>
+                  <PropertyGroup>
+                    <TestProperty>value</TestProperty>
+                  </PropertyGroup>
+
+                  <Target Name='Build'>
+                    <Message Text='Building test project'/>
+                  </Target>
+                </Project>
+                ";
+            string testDirectory = Path.GetTempPath();
+            string projectPath = Path.Combine(testDirectory, "ProjectGraphTest.proj");
+            File.WriteAllText(projectPath, projectContents);
```

Comment:
We have some nice methods to ensure cleanup of this type of test state. It's [`TestEnvironment`](https://github.com/Microsoft/msbuild/blob/master/src/Shared/UnitTests/TestEnvironment.cs).  Usage will probably be something like

```c#
using (var env = TestEnvironment.Create())
{
    TransientTestProjectWithFiles testProject = env.CreateTestProjectWithFiles(projectContents , Array.Empty<string>());
    var projectGraph = new ProjectGraph(testProject.ProjectFile);
}

```